### PR TITLE
*: display tiflash_write in cluster_info

### LINKED ISSUE #42064: Display tiflash_write in information_schema.cluster_info
## Enhancement
Display `tiflash_write` in information_schema.cluster_info for TiFlash cloud-native scenario.

### PR DESCRIPTION
**Pull request by hehechen**
### What problem does this PR solve?

Issue Number: close #42064

Problem Summary:
https://github.com/pingcap/tiflash/issues/7007
When the label of TiFlash contains `engine_role`:`write`, it means that this TiFlash is a `tiflash_write` node. So `tiflash_write` should be displayed in `information_schema.cluster_info`.
### What is changed and how it works?
Check the label and display `tiflash_write`.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
image
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/placement/common.go [hunk 1]
```diff
@@ -58,4 +58,8 @@ const (
 	// EngineLabelTiFlashCompute is for disaggregated tiflash mode,
 	// it's the lable of tiflash_compute nodes.
 	EngineLabelTiFlashCompute = "tiflash_compute"
+	// EngineRoleLabelKey is the label that indicates if the TiFlash instance is a write node.
+	EngineRoleLabelKey = "engine-role"
```
Comment (guo-shaoge) [inline ddl/placement/common.go:62]:
```suggestion
	EngineRoleLabelKey = "engine_role"
```

File: infoschema/tables.go [hunk 2]
```diff
@@ -1850,6 +1850,26 @@ func GetPDServerInfo(ctx sessionctx.Context) ([]ServerInfo, error) {
 	return servers, nil
 }
 
+func isTiFlashStore(store *metapb.Store) bool {
+	isTiFlash := false
+	for _, label := range store.Labels {
+		if label.GetKey() == placement.EngineLabelKey && label.GetValue() == placement.EngineLabelTiFlash {
+			isTiFlash = true
```
Comment (guo-shaoge) [inline infoschema/tables.go:1857]: break if true?

**Comment by guo-shaoge** [comment]
Other lgtm

File: infoschema/tables.go [hunk 3]
```diff
@@ -1850,6 +1850,24 @@ func GetPDServerInfo(ctx sessionctx.Context) ([]ServerInfo, error) {
 	return servers, nil
 }
 
+func isTiFlashStore(store *metapb.Store) bool {
+	for _, label := range store.Labels {
+		if label.GetKey() == placement.EngineLabelKey && label.GetValue() == placement.EngineLabelTiFlash {
```
Comment (breezewish) [inline infoschema/tables.go:1855]: Should we also check EngineLabelTiFlashCompute?

**Review by breezewish — APPROVED** [review]
The rest LGTM

File: infoschema/tables.go (hunk 3 above)
Comment (hehechen) [reply infoschema/tables.go:1855]: tiflash_compute is a store type in client-go, so no need to check here.

**Review by guo-shaoge — APPROVED** [review]

**Comment by breezewish** [comment]
This PR is now causing TiFlash add replica failures, and CLUSTER_CONFIG table missing TiFlash items:

```
mysql> alter database test set tiflash replica 1;
ERROR 1105 (HY000): the tiflash replica count: 1 should be less than the total tiflash server count: 0
mysql> select * from information_schema.cluster_info;
+-----------------+------------------+-------------------+----------------------------+------------------------------------------+----------------------+-----------------+-----------+
| TYPE            | INSTANCE         | STATUS_ADDRESS    | VERSION                    | GIT_HASH                                 | START_TIME           | UPTIME          | SERVER_ID |
+-----------------+------------------+-------------------+----------------------------+------------------------------------------+----------------------+-----------------+-----------+
| tidb            | 172.31.7.1:4000  | 172.31.7.1:10080  | 6.7.0-alpha                | 6f3122a1459f5cf361db1bb5e2ae5dd07214aff3 | 2023-03-15T04:05:01Z | 10m4.603312979s |   2342138 |
| tidb            | 172.31.7.2:4000  | 172.31.7.2:10080  | 6.7.0-alpha                | 6f3122a1459f5cf361db1bb5e2ae5dd07214aff3 | 2023-03-15T04:14:00Z | 1m5.603317463s  |   2306355 |
| tidb            | 172.31.7.1:4000  | 172.31.7.1:10080  | 6.7.0-alpha                | 6f3122a1459f5cf361db1bb5e2ae5dd07214aff3 | 2023-03-15T04:14:00Z | 1m5.60332026s   |   1925234 |
| pd              | 172.31.8.1:2379  | 172.31.8.1:2379   | 6.7.0-alpha                | c7227ff376cfbad6b3922cf7472ebd8669a387eb | 2023-03-15T04:13:41Z | 1m24.603321078s |         0 |
| tiflash_compute | 172.31.10.2:3930 | 172.31.10.2:20292 | 6.6.0-alpha-272-gb14adbfbe | b14adbfbe78cbe2a79c350d117780a10a5a3356e | 2023-03-15T04:14:05Z | 1m0.603321874s  |         0 |
| tiflash_write   | 172.31.9.1:3930  | 172.31.9.1:20292  | 6.6.0-alpha-272-gb14adbfbe | b14adbfbe78cbe2a79c350d117780a10a5a3356e | 2023-03-15T04:14:05Z | 1m0.603322565s  |         0 |
| tiflash_write   | 172.31.9.2:3930  | 172.31.9.2:20292  | 6.6.0-alpha-272-gb14adbfbe | b14adbfbe78cbe2a79c350d117780a10a5a3356e | 2023-03-15T04:14:05Z | 1m0.603323306s  |         0 |
| tikv            | 172.31.6.3:20160 | 172.31.6.3:20180  | 6.7.0-alpha                | c3e1cfb04046fe8ee1bc4a7ce453f273490e697a | 2023-03-15T04:13:47Z | 1m18.603324092s |         0 |
| tikv            | 172.31.6.1:20160 | 172.31.6.1:20180  | 6.7.0-alpha                | c3e1cfb04046fe8ee1bc4a7ce453f273490e697a | 2023-03-15T04:13:47Z | 1m18.603324801s |         0 |
| tikv            | 172.31.6.2:20160 | 172.31.6.2:20180  | 6.7.0-alpha                | c3e1cfb04046fe8ee1bc4a7ce453f273490e697a | 2023-03-15T04:13:47Z | 1m18.603328416s |         0 |
| tiflash_compute | 172.31.10.3:3930 | 172.31.10.3:20292 | 6.6.0-alpha-272-gb14adbfbe | b14adbfbe78cbe2a79c350d117780a10a5a3356e | 2023-03-15T04:14:05Z | 1m0.603329211s  |         0 |
| tiflash_compute | 172.31.10.1:3930 | 172.31.10.1:20292 | 6.6.0-alpha-272-gb14adbfbe | b14adbfbe78cbe2a79c350d117780a10a5a3356e | 2023-03-15T04:14:05Z | 1m0.60332996s   |         0 |
+-----------------+------------------+-------------------+----------------------------+------------------------------------------+----------------------+-----------------+-----------+
```

**Pull request merged**